### DISCUSSION TITLE
How to use pkg/tool/http?

### DISCUSSION THREAD

**Discussion by anjidevarasetty** [cuelang/cue#542] [General]
How can I use pkg/tool/http in cue file retrieve data from API and parse to update some variables in cue file itself.

```
get_request: {
	_get: http.Get & {
		url: "https://postman-echo.com/get?foo1=bar1"
		request: body: ""
		response: body: bytes
	}
    output: _get.response.body
}
```

https://cuelang.org/play/?id=WrBlawPAcss#cue@export@json

above is an example what I am trying. But I am getting below error

`get_request.output: cannot convert incomplete value "bytes" to JSON`

**Comment by verdverm** [comment]
the `tool/*` packages need to be used in `_tool.cue` files and run with `cue cmd`. In other words, you cannot eval/export like "pure" Cue.

See `cue cmd --help`

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/542.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
